substrate/frame/support/src/traits/tokens/fungibles/regular.rs

```diff
@@ -117,6 +117,10 @@ pub trait Inspect<AccountId>: Sized {
 
 	/// Returns `true` if an `asset` exists.
 	fn asset_exists(asset: Self::AssetId) -> bool;
+
+	/// Returns `Some(true)` if an `asset` exists and is sufficient. Returns `Some(false)`
+	/// if an `asset` exists, but is not sufficient. Returns `None` if an `asset` does not exist.
+	fn asset_sufficient(asset: Self::AssetId) -> Option<bool>;
```

Comment:
not sure if this concept is general enough, I cannot really say.
but an alternative can be a separate trait on assets pallet level, which lets you inspect some additional characteristics of an asset.
if you decide to continue in this direction, we need a good explanation here what sufficient means.
also the returned value can be just bool here, since you have `asset_exists` to check it is existence.